xmodule/capa/capa_problem.py

```diff
@@ -495,6 +505,31 @@ def get_grade_from_current_answers(self, student_answers):
 
         return newcmap
 
+    def get_grade_from_answers(
+        self, student_answers: dict, correct_map: dict
+    ) -> CorrectMap:
+        """
+        This method is based on `get_grade_from_current_answers` method but it
+        takes `student_answers` and `correct_map` as arguments instead of using
+        the ones stored in the problem.
+        """
+        newcmap = CorrectMap()
+        # Call each responsetype instance to do actual grading
+        for responder in self.responders.values():
+            # File objects are passed only if responsetype explicitly allows
+            # for file submissions.  But we have no way of knowing if
+            # student_answers contains a proper answer or the filename of
+            # an earlier submission, so for now skip these entirely.
+            # TODO: figure out where to get file submissions when rescoring.
+            if 'filesubmission' in responder.allowed_inputfields and student_answers is None:
+                _ = self.capa_system.i18n.gettext
+                raise Exception(_("Cannot rescore problems with possible file submissions"))
+
+            results = responder.evaluate_answers(student_answers, correct_map)
+            newcmap.update(results)
+
+        return newcmap
+
```

Comment:
is `get_grade_from_current_answers` still going to be used elsewhere? what's the difference between that method and this one?

Reply:
We could leave only one method since the two are very similar, adding an optional additional argument that receives a `correct_map`. If it is not sent, it takes the correct_map stored in `self.correct_map`